tests/evaluate/test_default_evaluator.py

```diff
@@ -2990,6 +2982,135 @@ def test_default_metrics_as_custom_metrics_static_dataset():
     assert "exact_match/v1" in results.metrics.keys()
 
 
+def test_derived_metrics():
+    def metric_1(predictions, targets, metrics):
```

Comment:
Do we disallow a custom metric called `metrics`?

Reply:
No, we don't disallow custom metrics (or such grading_context_columns) with names like "predictions", "targets", or "metrics" - we should.

Reply:
I think a warning is a good idea